torch/_tensor.py

```diff
@@ -132,7 +135,8 @@ def __deepcopy__(self, memo):
                 new_tensor.grad = self.grad.__deepcopy__(memo)
```

Comment:
The new_empty call should also have a check that `type(new_tensor) is type(self)` and an error that says: default implementation of deepcopy for non-wrapper subclass only works for subclass that implement new_empty and for which that function returns another instance of the same subclass. You can override `__deepcopy__` on your subclass if that is the intended behavior.